### PR TITLE
Render the sign-in/sign-out button in the Dashboard header using Ajax

### PR DESCRIPTION
**Pull request by philbogle**
We now render the user header containing the sign-in/sign-out button in the Dashboard header using Ajax.  This is the same Ajax action used for rendering the button in Pegasus.    Ajax rendering allows the user header to be rendered correctly for signed in users even in publicly cached pages.
### Testing
- [x] Visually compare positioning and rendering of Signin/Signout button on my machine vs. staging across a number dashboard pages.
- [x]  Run UI Tests against adhoc machine

### CODE REVIEW

**Comment by philbogle** [comment]
@Laurel This is the change to render the user header using ajax as we discussed.

File: dashboard/app/views/shared/_user_header_ajax.html.haml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#sign_in_or_user{:style=>'display:block; float: right'}
+:javascript
+  jQuery(document).ready(function() {
```
Comment (laurelfan) [inline dashboard/app/views/shared/_user_header_ajax.html.haml]: I think we usually use $ for jQuery

File: dashboard/app/views/shared/_user_header_ajax.html.haml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#sign_in_or_user{:style=>'display:block; float: right'}
```
Comment (laurelfan) [inline dashboard/app/views/shared/_user_header_ajax.html.haml]: can you add these styles to the stylesheet?

**Comment by laurelfan** [comment]
LGTM except for minor style comments

and it looks like some tests for the user menu should be changed to test ApiController#user_menu instead of HomeController#index

File: dashboard/app/views/shared/_user_header_ajax.html.haml (hunk 1 above)
Comment (philbogle) [inline dashboard/app/views/shared/_user_header_ajax.html.haml]: Done

File: dashboard/app/views/shared/_user_header_ajax.html.haml (hunk 2 above)
Comment (philbogle) [inline dashboard/app/views/shared/_user_header_ajax.html.haml]: Done.

**Comment by philbogle** [comment]
Laurel, please see whether the stylesheet changes I made look good.

I will check with you in person about the tests to update.

**Comment by philbogle** [comment]
I moved the appropriate tests to api_controller_test.  I also added tests to make sure the menu renders the correct "sign in" and "sign out" links based on the sign in state.

**Comment by philbogle** [comment]
I added back in the hack to fixed levels configured with an invalid base url (host name but no leading '//') so now all levels are rendering as expected.

**Comment by laurelfan** [comment]
LGTM. The UI test for the signin button that I just added may be relevant to this change:

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/features/signingIn.feature

I made it chrome only because other browsers were having problems with state between scenarios (the browser's cache I think?). It would be interesting to see if these things magically go away when it is ajaxified. It currently works on Chrome and Firefox.

**Comment by philbogle** [comment]
Eyes tests showed some layout issue and cross-test state contamination, so I'm going to roll this back for investigation.

**Pull request merged**